modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/actions/types.es.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export const EVENT_TYPES = {
+	CHANGE_ACTIVE_PAGE: 'activePageUpdated',
+	FIELD_BLUR: 'fieldBlurred',
+	FIELD_CHANGE: 'fieldEdited',
+	FIELD_DROP: 'fieldDrop',
+	FIELD_EVALUATED: 'evaluated',
+	FIELD_EVALUATION_ERROR: 'evaluationError',
+	FIELD_FOCUS: 'fieldFocused',
+	FIELD_REMOVED: 'fieldRemoved',
+	FIELD_REPEATED: 'fieldRepeated',
+	PAGE_ADDED: 'pageAdded',
+	PAGE_DELETED: 'pageDeleted',
+	PAGE_RESET: 'pageReset',
+	PAGE_SWAPPED: 'pagesSwapped',
+	PAGE_VALIDATION_FAILED: 'pageValidationFailed',
+	SUCCESS_CHANGED: 'successPageChanged',
+
+	// Internal events
+	/* eslint-disable sort-keys */
+
+	ALL: 'all',
+	UPDATE_PAGES: 'update_pages',
+
+	/* eslint-enable sort-keys */
+};
```

Comment:
Hmm yeah, good strategy. I was a little confused in naming this file because it did not have an `export default` and was also a constant, so could it be `eventTypes.js`?

Reply:
Sure. There is a pretty strong pattern of doing that in `layout-content-page-editor`; see [this directory](https://github.com/liferay/liferay-portal/tree/0ad5213737fced46e62e46d414a44b3bdc515a9d/modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/config/constants) which has many files like [this one](https://github.com/liferay/liferay-portal/blob/0ad5213737fced46e62e46d414a44b3bdc515a9d/modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/config/constants/itemTypes.js) (ie. file is called `itemTypes.js` and exports `ITEM_TYPES`).

I don't know why I said "default export" in my comment; I actually think it's fine for this to be a non-default export, so that people importing it always use the same name; eg:

```js
import {EVENT_TYPES} from './eventTypes';
```

The cases where I *would* want a default export would be things like classes, functions etc, because there the file name would be the exact same thing (case-sensitively), which is a strong push towards people using the right name even though the default doesn't force them to; eg:

```js
import WaffleMaker from './WaffleMaker';
import printCounterfeitMoney from './printCounterfeitMoney';
```

Reply:
I sent [a PR](https://github.com/liferay/liferay-frontend-guidelines/pull/169) to include this info in the guidelines. The rules are more complicated than I would like, but that's the reality of it for now.

Reply:
No problem sounds good. I'm going to go with `eventTypes`. Thanks